devs/docs/architecture.rst

```diff
@@ -0,0 +1,95 @@
+=====================
+CrateDB Architecture
+=====================
+
+This document assumes that you are familiar with using CrateDB. If you haven't
+used CrateDB before, please check out the `Getting Started
+<https://crate.io/docs/crate/getting-started/en/latest/>`_ section of the user
+documentation.
+
+CrateDB is a distributed SQL database. From a high-level perspective, the
+following components can be found in each node of CrateDB:
+
+.. image:: ../resources/architecture.svg
+
+Components
+-------------
+
+SQL Engine
+............
+
+CrateDB's heart consists of a SQL Engine which takes care of parsing SQL
+statements and executing them in the cluster.
+
+The engine is comprised of the following components:
+
+1. Parser: Breaks down the SQL statement into its components.
+2. Analyzer: Performs semantic processing of the statement including
+   verification, type annotation, and normalization.
+3. Planner: Builds an execution plan from the analyzed statement and optimizes it if possible.
+4. Executor: Executes the plan in the cluster and collects results.
+
+Class entry point:
+
+- `SqlParser`
+- `Analyzer`
+- `Planner`
+- `LogicalPlan` / `Plan`
+- `NodeOperationTree`
+- `ExecutionPhasesTask`
+- `BatchIterator`
+
+Input
+.....
+
+CrateDB accepts input from an HTTP/REST interface as well as from a Postgres
+compatible protocol. The Admin UI, crash and some of the connectors use the HTTP
```

Comment:
what about mqtt?

Reply:
Good point. I now mention it below that paragraph.